ci_feature/main.sh

```diff
@@ -196,16 +174,94 @@ if [ "$cAdvisorIsSecure" = true ] ; then
+            # should we default to container runtime docker?
+      else
+            containerRuntime=$(echo $podsResponse | jq -r '.items[0].status.containerStatuses[0].containerID' | cut -d ':' -f 1)
+            nodeName=$(echo $podsResponse | jq -r '.items[0].spec.nodeName')
+            export CONTAINER_RUN_TIME=$containerRuntime
+            export NODE_NAME=$nodeName
+            echo "configured container runtime on kubelet is : "$CONTAINER_RUN_TIME
+            echo "export CONTAINER_RUN_TIME="$CONTAINER_RUN_TIME >> ~/.bashrc
+            echo "export NODE_NAME="$NODE_NAME >> ~/.bashrc
+      fi
+
 else
       echo "Wget request using port 10250 failed. Using port 10255"
       export IS_SECURE_CADVISOR_PORT=false
       echo "export IS_SECURE_CADVISOR_PORT=false" >> ~/.bashrc
       export CADVISOR_METRICS_URL="http://$NODE_IP:10255/metrics"
       echo "export CADVISOR_METRICS_URL=http://$NODE_IP:10255/metrics" >> ~/.bashrc
+      echo "Making wget request to cadvisor endpoint with port 10255 to get the configured container runtime on kubelet"
+      podsResponse=$(wget -O- --server-response http://$NODE_IP:10255/pods)
+      if [ -z "$podsResponse" ]; then
+            echo "-e error  wget request to cadvisor endpoint /pods with port 10250 to get the configured container runtime on kubelet failed"
+            # should we default to container runtime docker?
+      else
+            containerRuntime=$(echo $podsResponse | jq -r '.items[0].status.containerStatuses[0].containerID' | cut -d ':' -f 1)
+            nodeName=$(echo $podsResponse | jq -r '.items[0].spec.nodeName')
+            export CONTAINER_RUN_TIME=$containerRuntime
+            export NODE_NAME=$nodeName
+            echo "configured container runtime on kubelet is : "$CONTAINER_RUN_TIME
+            echo "export CONTAINER_RUN_TIME="$CONTAINER_RUN_TIME >> ~/.bashrc
+            echo "export NODE_NAME="$NODE_NAME >> ~/.bashrc
+      fi
+
 fi
 
+# _total metrics will be available starting from k8s version 1.18 and current _docker_* and _runtime metrics will be deprecated
+# enable these when we add support for 1.18
+# export KUBELET_RUNTIME_OPERATIONS_TOTAL_METRIC="kubelet_runtime_operations_total"
+# echo "export KUBELET_RUNTIME_OPERATIONS_TOTAL_METRIC="$KUBELET_RUNTIME_OPERATIONS_TOTAL_METRIC >> ~/.bashrc
+# export KUBELET_RUNTIME_OPERATIONS_ERRORS_TOTAL_METRIC="kubelet_runtime_operations_errors_total"
+# echo "export KUBELET_RUNTIME_OPERATIONS_ERRORS_TOTAL_METRIC="$KUBELET_RUNTIME_OPERATIONS_ERRORS_TOTAL_METRIC >> ~/.bashrc
+
+# these metrics are avialble only on k8s versions <1.18 and will get deprecated from 1.18
+export KUBELET_RUNTIME_OPERATIONS_METRIC="kubelet_runtime_operations"
+export KUBELET_RUNTIME_OPERATIONS_ERRORS_METRIC="kubelet_runtime_operations_errors"
+
+#if container run time is docker then add omsagent user to local docker group to get access to docker.sock
+if [ "$CONTAINER_RUN_TIME" == "docker" ]; then
```

Comment:
Can you please do this all in a slightly different way so it minimizes the impact on (non-containerd) current clusters (which is the vast majority that we monitor now)? Basically switch/if class by comparing to runtime=="containerd" (instead of "docker" like you do here). This way if you are not able to determine if you are running in containerd, it will impact only containerd clusters (which is low right now, and will take sometime to pick up).

Reply:
set the default as "docker" and when aks moved to containerd then we can change the default.